website/docs/components/table/index.js

```diff
@@ -1,11 +1,28 @@
 import Component from '@glimmer/component';
+import { capitalize } from '@ember/string';
+import fetch from 'fetch';
 
 export default class Index extends Component {
   get STATES() {
     // these are used only for presentation purpose in the showcase
     return ['active', 'default', 'hover', 'focus'];
   }
-  //
-  // TODO! move also the data fetching here
-  //
+
+  async model() {
+    let response = await fetch('/api/folk.json');
+    console.log(response);
+    let { data } = await response.json();
+
+    // make sure the variable is declared outside of the loop
+    // so we can return it in the model response
+    let dataResponse = data.map((record) => {
+      let { id, attributes } = record;
+      return { id, ...attributes };
+    });
+    const keys = Object.keys(data[0].attributes);
+    const columns = keys.map((key) => {
+      return { key, label: capitalize(key) };
+    });
+    return { data: dataResponse, columns };
+  }
```

Comment:
Is the fundamental issue that we've lost the ability to have routers/controllers for specific pages of documentation?

Reply:
Yep, that's the cause. I was thinking I can do the fetch in this file, but it turns out to be quite twisty

Reply:
That's not ideal. I can think of a couple of ways to paper over the issue for this component but it's likely this will come up in the future as well 🤔

Reply:
Does the way alerts handle this work here as well? https://github.com/hashicorp/design-system/blob/main/website/docs/components/alert/index.js

Reply:
Yeah, any pre-declared data and functions work just fine, it's the asynchronicity of a fetch that puts it into trouble. have tried using `ember-concurrency` but it feels dirty to make the fetch request on `didInsert` or something..

Reply:
given we're really faking a request can we avoid the async aspect entirely and treat it as hardcoded data?

Reply:
yes, that's an option too, and probably one that helps us move forward

Reply:
@Dhaulagiri rebased to `main` and harcoded data, but still can't get it to render the 'model'-based tables in the showcase. I inspected the data object in both the old (dummy) and new website and they are identical to me 🤷

Reply:
It [turns out](https://github.com/hashicorp/design-system/pull/892) we were using an old component API for the table component 🤦